mutiny-core/src/lsp/lsps.rs

```diff
@@ -0,0 +1,228 @@
+
+                        if let Some(buy_response_sender) = pending_buy_requests.remove(&user_channel_id) {
+
+                            let invoice_expiry_delta_secs = 60 * 60 * 24 * 3;
+                            let (payment_hash, payment_secret) = self
+                                .channel_manager
+                                .create_inbound_payment(None, invoice_expiry_delta_secs, None)
+                                .unwrap();
+
+                            let lsp_route_hint = RouteHint(vec![RouteHintHop {
+                                src_node_id: counterparty_node_id,
+                                short_channel_id: scid,
+                                fees: RoutingFees { base_msat: 0, proportional_millionths: 0 },
+                                cltv_expiry_delta: cltv_expiry_delta.try_into().unwrap(),
+                                htlc_minimum_msat: None,
+                                htlc_maximum_msat: None,
+                            }]);
+
+                            let mut invoice = InvoiceBuilder::new(self.network.into())
+                                .description("Coins pls!".into())
+                                .payment_hash(sha256::Hash::from_slice(&payment_hash.0).unwrap())
+                                .payment_secret(payment_secret)
+                                .duration_since_epoch(utils::now())
+                                .min_final_cltv_expiry_delta(MIN_FINAL_CLTV_EXPIRY_DELTA.into())
+                                .private_route(lsp_route_hint);
+
+                            if let Some(payment_size_msat) = payment_size_msat {
+                                invoice = invoice.amount_milli_satoshis(payment_size_msat)
+                            };
+
+                            let invoice = invoice
+                                .build_signed(|hash| {
+                                    Secp256k1::new().sign_ecdsa_recoverable(
+                                        hash,
+                                        &self.keys_manager.get_node_secret_key(),
+                                    )
+                                })
+                                .unwrap();
+
+                            if let Err(_e) = buy_response_sender.send(invoice) {
+                                // log error
+                            }
+                        }
+                    },
+                    _ => {}
+                }
+            }
+
+            utils::sleep(1000).await;            
+        }
+    }
+
+    pub(crate) fn compute_opening_fee(&self, payment_size_msat: u64, opening_fee_min_fee_msat: u64, opening_fee_proportional: u64) -> Option<u64> {
+        payment_size_msat
+            .checked_mul(opening_fee_proportional)
+            .and_then(|f| f.checked_add(999999))
+            .and_then(|f| f.checked_div(1000000))
+            .map(|f| std::cmp::max(f, opening_fee_min_fee_msat))
+    }
```

Comment:
this function confuses me, adding docs to how it works would be nice

Reply:
yeah, added docs and opened a PR to lightning-liquidity to expose it -- it's part of the spec